packages/react-instantsearch-dom/src/core/createInstantSearchServer.js

```diff
@@ -180,5 +189,8 @@ export const findResultsState = function(App, props) {
     helper,
     sharedParameters,
     derivedParameters
-  );
+  ).then(results => {
+    results[0].metadata = metadata;
+    return results;
+  });
```

Comment:
IIRC this shape is opaque for the consumer of `react-instantsearch` i.e. we are the only once that interact with the value. In that case I would use a more appropriate structure (i.e. an object) rather than assigning the value to the first element of the array.

Reply:
that's an interesting option. I think I remember seeing people loop over the results since they always knew it's an array, but usually those use cases didn't really make sense to me

Reply:
https://github.com/algolia/react-instantsearch/pull/2977 here's that alternative